src/Program.cs

```diff
@@ -0,0 +1,59 @@
+﻿using CommandLine;
+using Newtonsoft.Json;
+using System.Net;
+using IP2Azure;
+
+Parser.Default.ParseArguments<Options>(args)
+    .WithParsed<Options>(opts =>
+        {
+            IPAddress? ip = null;
+            try {
+                ip = IPAddress.Parse(opts.Address);
+            }
+            catch (FormatException)
+            {
+                Console.WriteLine($"{opts.Address} is not a valid IPV4 address.");
+                Environment.Exit(1);
+            }
+            string json = System.IO.File.ReadAllText(opts.AzureJson);
+
+            // Map of IP Range (as string) to a list of services.
+            var servicesMap = JsonConvert.DeserializeObject<ServiceRanges>(json)?.GetServicesByRange();
+
+            if (servicesMap is null) {
+                Console.WriteLine($"Could not parse the JSON file {opts.AzureJson}.");
+                Environment.Exit(1);
+            }
+
+            // Emit a warning if non-IPv4 ranges are found in the JSON file.
+            var nonIpv4Count = servicesMap.Keys.Where(s => !IPRange.IsIpV4Range(s)).Count();
+            if (nonIpv4Count > 0 && !opts.Quiet)
+            {
+                Console.WriteLine($"WARNING: the JSON file contains {nonIpv4Count} non-IPv4 ranges, but this tool only supports IPv4.");
+            }
+
+            var ranges = servicesMap.Keys.Where(s => IPRange.IsIpV4Range(s)).Select(s => new IPRange(s));
```

Comment:
Would it make sense to sort these by most specific (smallest subnet) to least specific such that the output is ordered by most-likely service?

Reply:
Great suggestion, done!